src/common/providers/role-access.service.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@nestjs/common';
+import { AppConfigService } from '../../config/app/config.service';
+import { RoleAccess } from '../interfaces/role-access.interface';
+
+@Injectable()
+export class RoleAccessService {
+  public role: RoleAccess;
+
+  constructor(private appConfigService: AppConfigService) {
+    this.role = this.getRoleAccess();
+  }
+
+  getRoleAccess(): RoleAccess {
```

Comment:
```suggestion
  private getRoleAccess(): RoleAccess {
```
Secara default sifat nya akan public kang, baiknya di tambahin enkapsulasi nya

Reply:
Oh iya kang, method getRoleAccess nya ya,

Reply:
Oh iya baiknya this roles nya di assign di method nya aja kang, jangan di constructor, karena menurut yah roles memang di maping saat digunakan

Reply:
berarti kalo gini nanti tiap mau ngambil role harus jalanin method get value dari env nya dulu berarti ya kang?, kalo aku kenapa jalanin constructor karena biar si property role ini udah ke isi baik role admin maupun user, jadi kalo nanti butuh salahsatunya tinggal pake method getter aja kang satu-satu, mau get admin atau user, ide nya sih gitu. punten kang kalo ada masukan lagi.

Reply:
atau mungkin bisa juga tanpa harus buat interface RoleAccess, jadi bisa langsung jalanin method aja untuk ngambil role dari env nya kali ya, jadi gak ada encapsulasi nya

Reply:
> Kan udh di load di module config nya kang Sama aja sih mau di simpan di constructor mau engga beda nya saat assign nya aja, nah kalau constructor role ketika module di load langsung di assign ke method role, tapi kan role ini di panggil nya sewaktu waktu, kalau misal nya role ini harus bngt ada saat di load module nya, maka setuju di simpan di constructor,

iya kang kebutuhannya gak sering sih ya, kalo gini untuk interface RoleAccess nya di hapus aja berarti, karena gak digunakan juga sih kalo flownya seperti ini

Reply:
Coba rename jadi role() maka goalnya sama aja kan di ambil adalah isi value dan ekpektasi method yang kita buat, aku gak masalah untuk di pisah method biar rapi tapi enkapsulasi antara method internal dan mana yang nanti akan di pakai oleh class lain 😁

Reply:
oh iya boleh kang bisa nih, mangga kang untuk review kembali 😁